tfhe/src/integer/server_key/radix_parallel/sub.rs

```diff
@@ -257,4 +260,211 @@ impl ServerKey {
+                    // block in the ciphertext is >= 1, and that input borrow is either 0 or 1
+                    // so no overflow possible.
+                    crate::core_crypto::algorithms::lwe_ciphertext_sub_assign(
+                        &mut block.ct,
+                        &input_borrow.ct,
+                    );
+                    self.key.message_extract_assign(block);
+                });
+            assert!(ct.block_carries_are_empty());
+            // we know here that the result is a boolean value
+            // however the lut used has a degree of 2.
+            output_borrow.degree.0 = 1;
+            //assert!(output_borrow.degree.0 == 1);
+            (ct, output_borrow)
+        } else {
+            self.unsigned_overflowing_sub(lhs, rhs)
+        }
+    }
+
+    pub(super) fn generate_init_borrow_array(&self, sum_ct: &RadixCiphertext) -> Vec<Ciphertext> {
+        let modulus = self.key.message_modulus.0 as u64;
+
+        // This used for the first pair of blocks
+        // as this pair can either generate or not, but never propagate
+        let lut_does_block_generate_carry = self.key.generate_lookup_table(|x| {
+            if x < modulus {
+                1 // Borrows
+            } else {
+                0 // Does not borrow
+            }
+        });
+
+        let lut_does_block_generate_or_propagate = self.key.generate_lookup_table(|x| {
+            match x.cmp(&modulus) {
+                Ordering::Less => {
+                    1 // Generates borrow
+                }
+                Ordering::Equal => {
+                    2 // Propagates borrow
+                }
+                Ordering::Greater => {
+                    0 // No borrow generated or propagated
+                }
+            }
+        });
+
+        let mut generates_or_propagates = Vec::with_capacity(sum_ct.blocks.len());
+        sum_ct
+            .blocks
+            .par_iter()
+            .enumerate()
+            .map(|(i, block)| {
+                if i == 0 {
+                    // The first block can only output a borrow
+                    self.key
+                        .apply_lookup_table(block, &lut_does_block_generate_carry)
+                } else {
+                    self.key
+                        .apply_lookup_table(block, &lut_does_block_generate_or_propagate)
+                }
+            })
+            .collect_into_vec(&mut generates_or_propagates);
+
+        generates_or_propagates
+    }
+
+    pub(crate) fn compute_borrow_propagation_parallelized_low_latency(
+        &self,
+        generates_or_propagates: Vec<Ciphertext>,
+    ) -> (Vec<Ciphertext>, Ciphertext) {
+        let lut_borrow_propagation_sum = self
+            .key
+            .generate_lookup_table_bivariate(prefix_sum_borrow_propagation);
+
+        fn prefix_sum_borrow_propagation(msb: u64, lsb: u64) -> u64 {
+            if msb == 2 {
```

Comment:
small comment for the signification of the 2 or have an enum repr(u64) or whichever works

Reply:
Added an enum just like it's done for addition's carry propagation